docs/what-is-wrapped.md

```diff
@@ -0,0 +1,1007 @@
+object UserId extends Wrapped.Companion {
+  type Type = String
+  type Wrapper = UserId
+}
+
+final case class Age(value: Int) extends AnyVal with Wrapped[Int]
+object Age extends Wrapped.Companion {
+  type Type = Int
+  type Wrapper = Age
+}
+
+final case class Email(value: String) extends AnyVal with Wrapped[String]
+object Email extends Wrapped.Companion {
+  type Type = String
+  type Wrapper = Email
+}
+
+final case class User(
+  id: UserId,
+  age: Age,
+  email: Email
+)
+
+class UserRepository {
+  def get(id: UserId): User =
+    User(
+      id,
+      Age(21),
+      Email("user@email.com")
+    )
+}
+
+class UserService(repo: UserRepository) {
+  def get(id: UserId): User = repo.get(id)
+  
+  def sortByEmail(users: Seq[User]): Seq[User] = users.sortBy(_.email)
+  def sortByAge(users: Seq[User]): Seq[User] = users.sortBy(_.age)
+}
+```
+As you can see, our `Wrapped.Compaion` already looks similar to what we have in real code.
+One more time we reminded Scala that our custom classes are just thin wrappers on real types. <br>
+However, at this point it starts to look awkward. Introduction of custom wrappers allowed us to gain a lot of type safety.
+But at the same time we lost a lot of information and have to teach Scala how to work with them when it comes to trivial things like serialization or sorting.
+Sounds like we gain one feature and loose tons of features, which is unfair. Maybe we don't need it after all?<br>
+Yes, we do need it. We just need to make a fair comparison. Let's see what else wrappers can give us.
+
+_Note._ The next section is going to be a bit larger than previous two. However, this is the last thing that we'll need to understand real `Wrapped` implementation.
+So don't worry, it will be over soon. 
+
+Again, let's start with a simple example
+```scala
+final case class User(
+  id: String,
+  age: Int,
+  email: String
+)
+
+class UserRepository {
+  def get(id: String): User =
+    User(
+      id,
+      21,
+      "user@email.com"
+    )
+}
+
+class UserService(repo: UserRepository) {
+  def get(id: String): User = repo.get(id)
+}
+```
+Time passes, and our project grows. Now it can save users, send emails and provide useful information.
+```scala
+final case class User(
+  id: String,
+  age: Int,
+  email: String
+)
+
+object Utils {
+  private val dummyEmailRegexp: String = "(\\w|\\d)+@(\\w|\\d)\\.(\\w|\\d)"
+
+  def isValidAge(age: Int): Boolean = age >= 0 && age < 200
+  def isValidEmail(email: String): Boolean = email.matches(dummyEmailRegexp)
+}
+
+class EmailService {
+  def sendEmail(email: String, content: String): Unit = {
+    if (!Utils.isValidEmail(email)) throw new RuntimeException("email is invalid")
+    println(s"Sending [$content] to $email")
+  }
+}
+
+class LawsInfoService {
+  def canHaveRussianPassword(age: Int): Boolean = {
+    if (!Utils.isValidAge(age)) throw new RuntimeException("age is invalid")
+    age >= 14
+  }
+}
```

Comment:
I think there is a mistake here. method 'canHaveRussianPassword'. as I know, should not containt validation of age.

Reply:
Funny typo :) Fixed